lib/common/ipc_client.c

```diff
@@ -1238,21 +1244,37 @@ internal_ipc_get_reply(crm_ipc_t *client, int request_id, int ms_timeout,
             break;
         }
 
-        reply = pcmk__xml_parse(crm_ipc_buffer(client));
+        xml = pcmk__xml_parse(crm_ipc_buffer(client));
 
         if (hdr->qb.id < request_id) {
             crm_err("Discarding old reply %d (need %d)", hdr->qb.id, request_id);
-            crm_log_xml_notice(reply, "OldIpcReply");
-        } else {
+            crm_log_xml_notice(xml, "OldIpcReply");
+        } else if (hdr->qb.id > request_id) {
             crm_err("Discarding newer reply %d (need %d)", hdr->qb.id, request_id);
-            crm_log_xml_notice(reply, "ImpossibleReply");
+            crm_log_xml_notice(xml, "ImpossibleReply");
             pcmk__assert(hdr->qb.id <= request_id);
         }
-    } while (time(NULL) < timeout);
+    } while (time(NULL) < timeout || (timeout == 0 && *bytes == -EAGAIN));
 
-    if (*bytes < 0) {
+    if (*bytes > 0) {
+        crm_trace("Received %zd-byte reply %" PRId32 " to %s IPC %d: %.100s",
+                  *bytes, hdr->qb.id, client->server_name, request_id,
+                  crm_ipc_buffer(client));
+
+        if (reply != NULL) {
+            *reply = pcmk__xml_parse(crm_ipc_buffer(client));
+        }
+    } else if (*bytes < 0) {
         rc = (int) -*bytes; // System errno
+        crm_trace("No reply to %s IPC %d: %s " QB_XS " rc=%d",
```

Comment:
FWIW `QB_XS` doesn't matter for tracing. If we're at trace level, we're seeing gory detail